src/Model.php

```diff
@@ -26,7 +26,7 @@ class Model implements \JsonSerializable
     private static $automaticFormattingEnabled = true;
     private static $automaticValidationEnabled = false;
 
-    const SIGNED_32BIT_INT_MIN = -2147483648;
+    const SIGNED_32BIT_INT_MIN = -2147483647;
```

Comment:
What about this instead?

```suggestion
    const SIGNED_32BIT_INT_MIN = (int)-2147483648; # Cast to int to prevent 32-bit PHP from declaring this special number as a float
```